apps/archive/archive_correction.py

```diff
@@ -48,34 +48,54 @@ def on_update(self, updates, original):
+
         # updates for item in archive.
-        archive_item_updates = ({ITEM_STATE: CONTENT_STATE.CORRECTION, 'operation': CONTENT_STATE.CORRECTION}
-                                if not remove_correction
-                                else {ITEM_STATE: CONTENT_STATE.PUBLISHED, 'operation': ITEM_CANCEL_CORRECTION})
+        if not remove_correction:
+            archive_item_updates = {ITEM_STATE: CONTENT_STATE.CORRECTION, 'operation': CONTENT_STATE.CORRECTION}
+        elif remove_correction and archive_item.get('correction_sequence'):
+            archive_item_updates = {ITEM_STATE: CONTENT_STATE.CORRECTED, 'operation': 'correct'}
+        else:
+            archive_item_updates = {ITEM_STATE: CONTENT_STATE.PUBLISHED, 'operation': ITEM_CANCEL_CORRECTION}
 
         # updates for item in published.
-        published_item_updates = ({ITEM_STATE: CONTENT_STATE.BEING_CORRECTED,
-                                  'operation': CONTENT_STATE.BEING_CORRECTED} if not remove_correction
-                                  else {ITEM_STATE: CONTENT_STATE.PUBLISHED, 'operation': ITEM_CANCEL_CORRECTION})
+        if not remove_correction:
+            published_item_updates = {ITEM_STATE: CONTENT_STATE.BEING_CORRECTED,
+                                      'operation': CONTENT_STATE.BEING_CORRECTED}
+        elif remove_correction and published_article.get('correction_sequence'):
+            published_item_updates = {ITEM_STATE: CONTENT_STATE.CORRECTED, 'operation': 'correct'}
+        else:
+            published_item_updates = {ITEM_STATE: CONTENT_STATE.PUBLISHED, 'operation': ITEM_CANCEL_CORRECTION}
 
         # modify item in archive.
-        archive_service.system_update(archive_item.get(config.ID_FIELD),
-                                      archive_item_updates,
-                                      archive_item)
-        app.on_archive_item_updated(archive_item_updates,
-                                    archive_item, ITEM_CORRECTION)
+        archive_service.system_update(archive_item.get(config.ID_FIELD), archive_item_updates, archive_item)
+        app.on_archive_item_updated(archive_item_updates, archive_item, ITEM_CORRECTION)
 
         # modify item in published.
-        if original.get('state') == CONTENT_STATE.CORRECTED:
-            published_article = published_service.find_one(req=None,
-                                                           guid=original.get('guid'),
-                                                           correction_sequence=original.get('correction_sequence'))
-        else:
-            published_article = published_service.find_one(req=None,
-                                                           guid=original.get('guid'))
-
         published_service.patch(id=published_article.get(config.ID_FIELD), updates=published_item_updates)
 
+        #  if remove_correction, set the item values back to the old one.
```

Comment:
not sure what should be in the history, but might be better to just revert to previous version than doing this copy from published which has other metadata